### PR TITLE
Rename SecretKey to PrivateKey throughout identity.rs code base.

### PR DESCRIPTION
**Pull request by m-renaud**
# Description of change

Rename `SecretKey` to `PrivateKey` throughout the code base.

- Renames the `SecretKey` to `PrivateKey` in `identity-core/src/crypto/key/key.rs`
- Updates `PrivateKey` variable names from `secret`->`private` and `secret_key`->`private_key`
- Updates comments (unless I've missed some)
- Does **not** rename `Secret` to `Private` in `libjose/src/utils/crypto/key_repr.rs`
- Does **not** rename `MethodSecret` to `MethodPrivate` (can be done in a follow-up if desired)

Note that the `crypto` crate and various key signature algorithms within appear to use `Secret` and `SecretKey` to refer to the private part of the "secret" in some situations. For consistency I did not change any of the types or names in the `libjose` directory.

## Links to any relevant issues

#349

## Type of change

- [ ] Bug fix (a non-breaking change which fixes an issue)
- [ ] Enhancement (a non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation Fix

## How the change has been tested

`cargo build`, `cargo test` in both the root, examples, and `bindings/wasm/` directories.

Please let me know if there are other tests I should run.

## Change checklist

- [x] I have followed the contribution guidelines for this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Comment by m-renaud** [comment]
Updated comments, fixed breakages, and rebased changes into a new commit that is signed.

**Review by JelleMillenaar — APPROVED** [review]
LGTM, just renaming and a few comments that changed.

File: bindings/wasm/src/crypto/key_collection.rs [hunk 1]
```diff
@@ -133,7 +133,7 @@ impl KeyCollection {
 
     let iter: _ = data.keys.iter().flat_map(|data| {
       let pk: PublicKey = decode_b58(&data.public).ok()?.into();
-      let sk: SecretKey = decode_b58(&data.secret).ok()?.into();
+      let sk: PrivateKey = decode_b58(&data.private).ok()?.into();
 
       Some((pk, sk))
```
Comment (cycraig) [inline bindings/wasm/src/crypto/key_collection.rs:138]: Probably worth using `public_key/private_key` or `public/private` variable names here too (GitHub doesn't allow posting a suggestion on "deleted lines" unfortunately).

I imagine the `pk/pk` naming conflict is why people started using `Secret` (`sk`) instead.

File: identity-core/src/crypto/signature/traits/core.rs [hunk 2]
```diff
@@ -13,13 +13,13 @@ use crate::error::Result;
 /// A common interface for digital signature creation.
 pub trait Sign {
   /// The secret key type of this signature implementation.
```
[the remark this answers is not in the document]
Comment (m-renaud) [reply identity-core/src/crypto/signature/traits/core.rs:15]: Resolved locally.

File: bindings/wasm/src/crypto/key_collection.rs (hunk 1 above)
Comment (m-renaud) [reply bindings/wasm/src/crypto/key_collection.rs:138]: Resolved locally.

**Review by cycraig — APPROVED** [review]

**Pull request merged**